File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid.Core/Component/ControlAttachment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Support.V4.Content;
using Android.Support.V7.Widget;
using Android.Util;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Bean;
using BPMOPMobile.Droid.Core.Class;
using BPMOPMobile.Droid.Core.Common;
using Java.IO;

namespace BPMOPMobile.Droid.Core.Component
{
    public class ControlAttachment : ControlBase
    {
        private LinearLayout _parentView { get; set; }
        private ViewElement _element { get; set; }
        private RecyclerView _recyAttachment;
        public int _widthScreenTablet = -1;
        List<KeyValuePair<string, string>> _lstAttachment = new List<KeyValuePair<string, string>>();

        public ControlAttachment(Activity _mainAct, LinearLayout _parentView, ViewElement _element, int _widthScreenTablet) : base(_mainAct)
        {
            this._mainAct = _mainAct;
            this._parentView = _parentView;
            this._element = _element;
            this._widthScreenTablet = _widthScreenTablet;
            InitializeComponent();
        }
        public override void InitializeComponent()
        {
            base.InitializeComponent();
            _recyAttachment = new RecyclerView(_mainAct);

            //if (_lnContent != null)
            //{
            //    _lnContent.Click += HandleTouchDown;
            //}
        }
        public override void InitializeFrameView(LinearLayout frame)
        {
            if (_element.Hidden == true) // Check xem có ẩn view hay không
                return;

            base.InitializeFrameView(frame);
            _tvValue.Visibility = ViewStates.Gone;
            _lnLine.Visibility = ViewStates.Gone;
            _lnContent.RemoveView(_lnLine); // Remove line ra để add lại sau

            _lnContent.LayoutParameters = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MatchParent, LinearLayout.LayoutParams.WrapContent);



            frame.AddView(_recyAttachment);
            frame.AddView(_lnLine);

        }
        public override void SetTitle()
        {
            base.SetTitle();

            _tvTitle.Text = _element.Title;

            if (_element.IsRequire && _element.Enable)
            {
                _tvTitle.Text += " (*)";
                CmmDroidFunction.SetTextViewHighlightControl(_mainAct, _tvTitle);
            }
        }
        public override void SetValue()
        {
            base.SetValue();
            var data = _element.Value.Trim();
            if (data.Contains(";#"))
            {
                var _arrAttachment = data.Split(new string[] { ";#" }, StringSplitOptions.None);
                if (_arrAttachment.Length > 2)
                {
                    for (var i = 0; i < _arrAttachment.Length; i += 2)
                    {
                        KeyValuePair<string, string> item = new KeyValuePair<string, string>(_arrAttachment[i], _arrAttachment[i + 1]);
                        _lstAttachment.Add(item);
                    }
                }
                else
                    _lstAttachment.Add(new KeyValuePair<string, string>(_arrAttachment[0], _arrAttachment[1]));
            }

            StaggeredGridLayoutManager staggeredGridLayoutManager = new StaggeredGridLayoutManager(1, LinearLayoutManager.Horizontal);
            AdapterControlAttachmentHorizontal _adapterAttachFile = new AdapterControlAttachmentHorizontal(_mainAct, _lstAttachment, _widthScreenTablet);
            _adapterAttachFile.ItemClick += Click_ItemAttach;
            _recyAttachment.SetLayoutManager(staggeredGridLayoutManager);
            _recyAttachment.SetAdapter(_adapterAttachFile);
        }
        /// <summary>
        /// Event khi click vào Item
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void Click_ItemAttach(object sender, int e)
        {
            MinionActionCore.OnElementClickAttachmentDetailWorkflow(null, new MinionActionCore.ElementAttachFileClick(_element, _lstAttachment[e]));
        }
    }
    public class AdapterControlAttachmentHorizontal : RecyclerView.Adapter
    {
        public event EventHandler<int> ItemClick;
        public Activity _mainAct;
        public List<KeyValuePair<string, string>> _lstAttachment;
        public int _widthScreenTablet = -1;

        public AdapterControlAttachmentHorizontal(Activity _mainAct, List<KeyValuePair<string, string>> _lstAttachment, int _widthScreenTablet)
        {
            this._mainAct = _mainAct;
            this._lstAttachment = _lstAttachment;
            this._widthScreenTablet = _widthScreenTablet;
        }
        private void OnItemClick(int obj)
        {
            ItemClick(this, obj);
        }
        public override int ItemCount => _lstAttachment.Count;
        public override RecyclerView.ViewHolder OnCreateViewHolder(ViewGroup parent, int viewType)
        {
            View itemView = LayoutInflater.From(parent.Context).Inflate(Resource.Layout.ItemControlAttachmentHorizontal, parent, false);
            int _widthRow;
            if (_widthScreenTablet == -1) // ForPhone
            {
                DisplayMetrics dm = _mainAct.Resources.DisplayMetrics;
                _widthRow = dm.WidthPixels / 3;
            }
            else // For Tablet
            {
                _widthRow = _widthScreenTablet / 3;
            }
            itemView.LayoutParameters = new ViewGroup.LayoutParams(_widthRow, ViewGroup.LayoutParams.WrapContent);

            ControlAttachmentHorizontal holder = new ControlAttachmentHorizontal(itemView, OnItemClick);
            return holder;
        }
        public override void OnBindViewHolder(RecyclerView.ViewHolder holder, int position)
        {
            ControlAttachmentHorizontal vh = holder as ControlAttachmentHorizontal;
            KeyValuePair<string, string> _currentAttachment = _lstAttachment[position];
            if (!String.IsNullOrEmpty(_currentAttachment.Value))
            {
                vh._tvFileName.Text = _currentAttachment.Value;
            }
            else
            {
                vh._tvFileName.Text = "";
            }
            if (_currentAttachment.Key == "1" || _currentAttachment.Key == "3") // để hình pdf
            {
                //_img.Image = UIImage.FromFile("Icons/icon_attachment_pdf.png");
            }
            else // hình khác
            {

            }
        }
    }
    public class ControlAttachmentHorizontal : RecyclerView.ViewHolder
    {
        public ImageView _imgAvatar { get; set; }
        public TextView _tvFileName { get; set; }
        public ControlAttachmentHorizontal(View itemview, Action<int> listener) : base(itemview)
        {
            _imgAvatar = itemview.FindViewById<ImageView>(Resource.Id.img_ItemControlAttachmentHorizontal_Avatar);
            _tvFileName = itemview.FindViewById<TextView>(Resource.Id.tv_ItemControlAttachmentHorizontal_FileName);
            itemview.Click += (sender, e) => listener(base.LayoutPosition);
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/core/component/ControlAttachment.java
package com.vuthao.bpmop.core.component;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.util.DisplayMetrics;
import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import com.vuthao.bpmop.R;
import com.vuthao.bpmop.base.Functions;
import com.vuthao.bpmop.base.model.dynamic.ViewElement;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ControlAttachment extends ControlBase implements ControlBase.ControlAttachmentHorizontalListener {
    private final ViewElement element;
    private RecyclerView recyAttachment;
    public int widthScreenTablet = -1;
    private final ArrayList<Pair<String, String>> lstAttachment = new ArrayList<Pair<String, String>>();

    public ControlAttachment(Activity mainAct, LinearLayout parentView, ViewElement element, int widthScreenTablet)
    {
        super(mainAct);
        this.mainAct = mainAct;
        this.element = element;
        this.widthScreenTablet = widthScreenTablet;
        initializeComponent();
    }

    @Override
    public void initializeComponent() {
        super.initializeComponent();
        recyAttachment = new RecyclerView(mainAct);
    }

    @Override
    public void initializeFrameView(LinearLayout frame) {
        if (element.isHidden()) return;
        super.initializeFrameView(frame);

        tvValue.setVisibility(View.GONE);
        lnLine.setVisibility(View.GONE);
        lnContent.removeView(lnLine);
        lnContent.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT));

        frame.addView(recyAttachment);
        frame.addView(lnLine);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void setTitle() {
        super.setTitle();

        tvTitle.setText(element.getTitle());
        if (element.isRequire() && element.isEnable()) {
            tvTitle.setText(tvTitle.getText() + " (*)");
            Functions.share.setTVHighlightControl(mainAct, tvTitle);
        }
    }

    @Override
    public void setValue() {
        super.setValue();

        String data = element.getValue().trim();
        if (data.contains(";#")) {
            String[] arrAttachment = data.split(";#");
            if (arrAttachment.length > 2) {
                for (int i = 0; i < arrAttachment.length; i++) {
                    Pair<String, String> item = new Pair<>(arrAttachment[i], arrAttachment[i + 1]);
                    lstAttachment.add(item);
                }
            } else {
                lstAttachment.add(new Pair<>(arrAttachment[0], arrAttachment[1]));
            }
        }

        StaggeredGridLayoutManager staggeredGridLayoutManager = new StaggeredGridLayoutManager(1, LinearLayout.HORIZONTAL);
        ControlAttachmentHorizontalAdapter _adapterAttachFile = new ControlAttachmentHorizontalAdapter(this, mainAct, lstAttachment, widthScreenTablet);
        recyAttachment.setLayoutManager(staggeredGridLayoutManager);
        recyAttachment.setAdapter(_adapterAttachFile);
    }

    @Override
    public void OnClick(int pos) {
        // attach Click to detail
    }

    public class ControlAttachmentHorizontalAdapter extends RecyclerView.Adapter<ControlAttachmentHorizontalAdapter.ControlAttachmentHorizontalHolder>  {
        public Activity mainAct;
        public ArrayList<Pair<String, String>> lstAttachment;
        public int widthScreenTablet = -1;

        public ControlAttachmentHorizontalAdapter(ControlAttachmentHorizontalListener listener, Activity mainAct, ArrayList<Pair<String, String>> lstAttachment, int widthScreenTablet) {
            this.mainAct = mainAct;
            this.lstAttachment = lstAttachment;
            this.widthScreenTablet = widthScreenTablet;
        }

        @NonNull
        @Override
        public ControlAttachmentHorizontalHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_control_attachment_horizontal, parent, false);
            int widthRow;
            if (widthScreenTablet == -1) {
                DisplayMetrics dm = mainAct.getResources().getDisplayMetrics();
                widthRow = dm.widthPixels / 3;
            } else {
                widthRow = widthScreenTablet / 3;
            }

            view.setLayoutParams(new ViewGroup.LayoutParams(widthRow, ViewGroup.LayoutParams.WRAP_CONTENT));

            return new ControlAttachmentHorizontalHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ControlAttachment.ControlAttachmentHorizontalAdapter.ControlAttachmentHorizontalHolder holder, int position) {
            Pair<String,String> currentAttachment = lstAttachment.get(position);
            if (!Functions.isNullOrEmpty(currentAttachment.second)) {
                holder.tvFileName.setText(currentAttachment.second);
            } else {
                holder.tvFileName.setText("");
            }
        }

        @Override
        public int getItemCount() {
            return lstAttachment.size();
        }

        public class ControlAttachmentHorizontalHolder extends RecyclerView.ViewHolder {
            @BindView(R.id.img_ItemControlAttachmentHorizontal_Avatar)
            ImageView imgAvatar;
            @BindView(R.id.tv_ItemControlAttachmentHorizontal_FileName)
            TextView tvFileName;
            public ControlAttachmentHorizontalHolder(@NonNull View itemView) {
                super(itemView);
                ButterKnife.bind(this, itemView);
            }
        }
    }
}
